carmin.yaml

```diff
@@ -532,61 +532,8 @@ components:
           type: string
     Pipeline:
       type: object
-      required:
-        - identifier
-        - name
-        - version
-        - properties
-        # - parameters optional in 0.2
-      properties:
-        identifier:
-          type: string
-          format: ascii
-        name:
-          type: string
-        version:
-          type: string
-          format: ascii
-        description:
-          type: string
-        canExecute:
-          type: boolean
-          description: true if the user who requested the pipeline can execute it
-        parameters:
-          type: array
-          items:
-            $ref: '#/components/schemas/PipelineParameter'
-        properties:
-          type: object
-          additionalProperties:
-            type: string
-          description: the properties (as keys) and their values that describe the pipeline. 
-            The properties used must be listed in the "supportedPipelineProperties" of the "getPlatformProperties" method.
-        errorCodesAndMessages:
-          type: array
-          items:
-            $ref: '#/components/schemas/ErrorCodeAndMessage'
-    PipelineParameter:
-      type: object
-      required:
-        - name
-        - type
-        - isOptional
-        - isReturnedValue
-      properties:
-        name:
-          type: string
-          format: ascii
-        type:
-          $ref: '#/components/schemas/ParameterType'
-        isOptional:
-          type: boolean
-        isReturnedValue:
-          type: boolean
-        defaultValue:
-          description: Default value. It must be consistent with the parameter type.
-        description:
-          type: string
+      $ref: 'https://raw.githubusercontent.com/boutiques/boutiques/master/tools/python/boutiques/schema/descriptor.schema.json'
```

Comment:
Many aspect of the Boutique specification are related to the implementation of the pipeline (e.g. environment variables, how to create a command line for executing the pipeline) but are not related to the web services for calling the pipeline. In my opinion CARMIN should be implementation independent. However, we may add some implementation related web services to ask for information on a specific implementation. For instance, a client would ask via CARMIN to get a Boutique specification for a Pipeline.

Reply:
Note that most of these properties, including environment variables, container images, etc are not required in the Boutiques descriptor. That is, it is ok for a descriptor to just contain inputs and outputs, see minimal descriptor above. So platforms wouldn't have to use all those features.

How about we make the CARMIN schema such that additional properties can be added? Such that a Boutiques object is a valid CARMIN pipeline but a valid CARMIN pipeline doesn't have to have to have any additional property that aren't in the current version of CARMIN? In the minimal descriptor above, making 'command-line' and 'schema-version' optional would do it. Then, some platforms may return CARMIN descriptors that happen to be valid Boutiques descriptors and others might return CARMIN descriptors that are not valid Boutiques descriptors. What do you think?

Reply:
Proposed modifications following discussion above:
* in Boutiques (breaks backward compatibility):
  - add `id` field to tool.
  - rename `tool-version` and `output-files`.
  - add `type` to output.
* in CARMIN:
   - in pipeline
       - split parameters in `inputs` and `outputs`. 
       - allow additional properties.
       - make properties optional and rename them to `tags`.
       - rename `errorCodeAndMessages` to `error-codes`, `errorCode` to `code` and `errorMessage` to `description`.
   - in inputs and outputs
       - allow additional properties     
       - rename `isOptional` to `optional`
       - adopt Boutiques types (see table above)

The following example is a valid Boutiques tool and a valid CARMIN pipeline, according to the modifications proposed above:
```
{
  "id": "echo", // breaks backward compatibility in Boutiques
  "name": "echo",
  "version": "1.0", // breaks backward compatibility in Boutiques
  "description": "A simple script to test output files",
  "command-line": "echo [PARAM] > output.txt", // additional property
  "schema-version": "0.5", // additional property
  "inputs": [{
      "id": "param", // additional property
      "name": "Parameter",
      "value-key": "[PARAM]", // additional property
      "type": "Number"
  }],
  "outputs": [{  // breaks backward compatibility in Boutiques
      "id": "output_file",
      "type": "File", // breaks backward compatibility in Boutiques
      "name": "Output file",
      "path-template": "output.txt" // additional property
  }]
}
```

Reply:
@sapetnioc @axlbonnet How about going down the multiple-method route suggested by @sapetnioc above? We could just have an option of `GET pipeline/<id>`, for instance `GET pipeline/<id>&format=boutiques` (format being optional of course), and add a property to platform properties that lists the list of supported formats? 

Advantages:
- backward compatibility with CARMIN and Boutiques
- simplicity to implement

I'm very tempted by this option now.